Assets/Scripts/AStar/AStarExampleTilemap.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using AStar;
+using UnityEngine.Tilemaps;
+
+public class AStarExampleTilemap : MonoBehaviour
+{
+    private NodeMap nodeMap = new NodeMap(0.5f, 0.5f, 0, 50, 50, 0.2f); 
+    /*
+     * For Astar to work there needs to be a map for it to traverse
+     * 
+     * the parameters (in order): Origin x, Origin y, Origin z, # of x nodes, # of y nodes, radius of each nodes
+     * 
+     * Origin start at bottom left
+     */
+
+
+    // Path finder's target position to reach
+    public Transform endPosition;
+
+    // Tile map for untraversable tiles
+    public Tilemap wallTileMap;
+
+    // Path finder returns a Path object which contains an array of Node objects representing the waypoints the path finder took to get to target position
+    Path pathToEnd = new Path(new NodeMap(0.5f, 0.5f, 0, 50, 50, 0.2f), new Node[0]);
+
+    // Variables for a cool down system to constantly update path
+    float timeMemory = 3f;
+    public float pathUpdateTimer = 3f;
+
+
+    // How many cycles to loop through before giving up on looking for a path to target position
+    public int searchCycles = 2000;
+
+    private void Start()
+    {
+        // To "Bake" the node map to capture untraversable areas. Whill scan the each node on wallTileMap for untraversable tiles on nodeMap
+        nodeMap.BakeBlockedMap(wallTileMap);
+
+        // Baking traversable but not desirable areas with 1.5 cost
+        nodeMap.BakeCostMap(LayerMask.GetMask("Water"), 1.5f);
+
+        /* 
+         * For 3D map baking
+         * 
+         * Overloaded method, takes integer repersenting LayerMask id for *IGNORED* physics layer
+         * 
+         * Scans each node with Physics.CheckSphere() for colliders around node, any collider detected not in above mentioned LayerMask will be consider an obstacle.
+         * Therefore, marking that node as untraversable.
+         */
+
+        // nodeMap.BakeBlockedMap(LayerMask.GetMask("Ignore Raycast"));
+
+
+        // Save baked map as "MyNodeMap.bin" at ../UnityProjectFile/Assets/MyNodeMap.bin
+        nodeMap.Save("MyNodeMap");
+
+        // Load baked map from ../UnityProjectFile/Assets/MyNodeMap.bin
+        nodeMap = NodeMap.Load("MyNodeMap");
+    }
+
+    private void Update()
+    {
+        timeMemory += Time.deltaTime;
```

Comment:
This isn't something you're obligated to do, but consider using coroutines - You can achieve the same timer effect with much cleaner code.